include/iso15118/d20/session.hpp

```diff
@@ -99,6 +99,10 @@ class Session {
         return selected_services.selected_control_mode;
     }
 
+    message_20::MobilityNeedsMode get_selected_mobility_needs_mode() const {
+        return selected_services.selected_mobility_needs_mode;
+    }
+
```

Comment:
Rather than _bloating_ the interface of session here, just have a single function which returns a `const &` to the `selected_services`?

Reply:
We can also do that

Reply:
Done